### PR TITLE
21-Add address1 address2 city state and zip to company

### PR DESCRIPTION
**Pull request by godfreym**
Issue #21 

## Story
AS an admin 
I WANT to add the address1, address2, city, state and zip as part of company's address
SO THAT I can track the complete info regarding the companies addresses

## Sample Format
```
700 Craig-head St <address1>
Suite 106 <address2>
Nashville, TN 37204 <city>, <state> <zip>
```

### CODE REVIEW

**Comment by godfreym** [comment]
@leonardk1 Please Review

**Comment by arafatm** [comment]
@NashvilleJim Do you really need to split the address this way? Seems like unnecessary split. I'd just put it in a single text box. You can still search/validate etc instead of having to type each field in separately.

File: app/models/company.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@ class Company < ActiveRecord::Base
   validates :title,           presence: true
   validates :description,     presence: true
   validates :about,           presence: true
-  validates :address,         presence: true
+  validates :address1,        presence: true
+  validates :city,            presence: true
+  validates :state,           presence: true
```
Comment (arafatm) [inline app/models/company.rb]: @godfreym american states are 2 capital letters. you should be able to find a list so you can make a dropdown.

File: app/models/company.rb [hunk 2]
```diff
@@ -6,6 +6,9 @@ class Company < ActiveRecord::Base
...
+  validates :zip,             presence: true
```
Comment (arafatm) [inline app/models/company.rb]: @godfreym zip codes are 5 digits.

File: app/controllers/companies_controller.rb [hunk 3]
```diff
@@ -56,8 +56,8 @@ def destroy
   private
 
   def company_params
-    params.require(:company).permit(:name, :title, :description, :about, :address, :facebook, 
-    	:google_plus, :twitter, :linkedin, :dribble, :logo, :logo_cache, :intro_image,
-      :intro_image_cache)
+    params.require(:company).permit(:name, :title, :description, :about, :address1, :address1, :city,
```
Comment (arafatm) [inline app/controllers/companies_controller.rb]: @godfreym is that 2nd address1 supposed to be address2?

File: app/views/companies/index.html.erb [hunk 4]
```diff
@@ -18,6 +18,8 @@
 	            <th class="text-center">Company Name</th>
 	            <th class="text-center">Title</th>
 	            <th class="text-center">Address</th>
+	            <th class="text-center">City</th>
+	            <th class="text-center">State & Zip</th>
```
Comment (arafatm) [inline app/views/companies/index.html.erb]: @godfreym for consistency, split these into 2 columns.

File: app/models/company.rb (hunk 2 above)
Comment (godfreym) [inline app/models/company.rb]: @arafatm Zip Codes have extensions, meaning they can be 12345-6789 too

File: app/controllers/companies_controller.rb (hunk 3 above)
Comment (godfreym) [inline app/controllers/companies_controller.rb]: @arafatm Yes, Fixed.. I need :eyeglasses:

File: app/views/companies/index.html.erb (hunk 4 above)
Comment (godfreym) [inline app/views/companies/index.html.erb]: @arafatm Fixed this

File: app/models/company.rb (hunk 2 above)
Comment (arafatm) [inline app/models/company.rb]: @godfreym no one uses the extensions except for the post office. 5 digits is standard

Comment (godfreym) [inline app/models/company.rb]: @arafatm OK Understood, I fixed this by validating for 5 digits

File: app/models/company.rb (hunk 1 above)
Comment (godfreym) [inline app/models/company.rb]: @arafatm Added a new validation to make sure only 2 capital letters are used for state

File: app/models/company.rb [hunk 5]
```diff
@@ -2,10 +2,15 @@ class Company < ActiveRecord::Base
...
-  validates :address,         presence: true
+  validates :name,        presence: true
+  validates :title,       presence: true
+  validates :description, presence: true
+  validates :about,       presence: true
+  validates :address1,    presence: true
+  validates :city,        presence: true
+  validates :state,       presence: true
+  validates :zip,         presence: true, length: { is: 5 }
+  validates :state,       presence: true, length: { is: 2 }, format: { with: /\A[A-Z]+\z/,
```
Comment (godfreym) [inline app/models/company.rb]: @arafatm New validations added, Please Review

File: app/models/company.rb (hunk 1 above)
Comment (godfreym) [inline app/models/company.rb]: @arafatm I have made a drop down for all states

**Comment by godfreym** [comment]
@arafatm Responded to all comments and we now have a drop down for states, Please Review

**Comment by arafatm** [comment]
:+1:

**Pull request merged**